### PR TITLE
Address Identity API review (10/16)

### LINKED ISSUE #5885: Rename deviceCodeChallengeConsumer to challengeConsumer on DeviceCodeCredentialBuilder

### PR DESCRIPTION
**Pull request by jianghaolu**
Resolves #5885 #5886

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java [hunk 1]
```diff
@@ -58,7 +57,7 @@
                 return Mono.empty();
             }
         }).switchIfEmpty(
-            Mono.defer(() -> identityClient.authenticateWithDeviceCode(request, deviceCodeChallengeConsumer)))
+            Mono.defer(() -> identityClient.authenticateWithDeviceCode(request, challengeConsumer)))
```
Comment (conniey) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java]: It's possible to have a null challengeConsumer. The calling code may throw a nullpointerexception.

Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java]: It's not possible through the validation in the builder:
```java
    public DeviceCodeCredential build() {
        ValidationUtil.validate(getClass().getSimpleName(), new HashMap<String, Object>() {{
                put("clientId", clientId);
                put("challengeConsumer", challengeConsumer);
            }});
        return new DeviceCodeCredential(clientId, tenantId, challengeConsumer, identityClientOptions);
    }
```

Comment (conniey) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java]: Ahh! I see the ValidationUtil now. Thanks. :)

**Comment by jianghaolu** [comment]
Approved by adpship.

**Pull request merged**